stdlib/LinearAlgebra/test/tridiag.jl

```diff
@@ -665,5 +665,25 @@ using .Main.ImmutableArrays
     @test convert(AbstractArray{Float64}, Tsym)::SymTridiagonal{Float64,ImmutableArray{Float64,1,Array{Float64,1}}} == Tsym
     @test convert(AbstractMatrix{Float64}, Tsym)::SymTridiagonal{Float64,ImmutableArray{Float64,1,Array{Float64,1}}} == Tsym
 end
+    
+@testset "dot(x,A,y) for A::Tridiagonal or SymTridiagonal" begin
+    for elty in (Float32, Float64, ComplexF32, ComplexF64, Int)
+        n = 4
+        v1 = rand(elty, n-1)
+        v2 = rand(elty, n)
+        v3 = rand(elty, n-1)
+        T = Tridiagonal(v1, v2, v3)
+        Tsym = SymTridiagonal(v2, v1)
+        x = fill(convert(elty, 1), n)
+        y = fill(convert(elty, 2), n)   
+        x1 = fill(convert(elty, 1), 0)
+        T1 = Tridiagonal(x1, x1, x1)
+        Tsym1 = SymTridiagonal(x1, x1)
+        @test dot(x, T, y)  ≈ dot(x, T*y)
+        @test dot(x, Tsym, y)  ≈ dot(x, Tsym*y)   
+        @test dot(x1, T1, x1)   ≈ 0.0
+        @test dot(x1, Tsym1, x1)   ≈ 0.0   
+    end
+end
```

Comment:
3-arg `dot` is covered many times, except for the case when the vector is empty, i.e., your last two lines. I suggest to reduce this testset to those tests, to avoid increasing testing time unnecessarily.